### PR TITLE
Fix usage of direct memberships based on an organization rights

### PR DESCRIPTION
**Pull request by nicholaspcr**
#### Summary

References https://github.com/TheThingsIndustries/lorawan-stack-support/issues/927
Which in short, refers to a bug where organization API keys didn't have their rights validated properly when fetching entities. Always showing only the safe for public fields in an entity, in this case the gateways information.

#### Changes

- Update `FindAccountMembershipChains` method to validate direct membership type to set the appropriate Identifier. 

#### Testing

existing UT and manual testing.

##### Regressions

This is a bug fix.

It does change behaviors, allowing API Keys with the proper rights to access non public safe information. As this is allowed by design and restricted by the rights implementation, plus is the previous behavior before removing the gormstore.

So it affects the interaction of users using API Keys of organizations and what they can do with other entities, restoring what they could do `v3.21` backwards.

#### Notes for Reviewers

From looking into how it was previously implemented on the gormstore ( see bellow ) and that `direct` and `indirect` chains have been separated into their own `membershipChain`. The only thing conditional missing to attain the previous behavior that existed with `gormstore` is the addition of the `accountType` validation.

```go
func (m membershipChain) GetMembershipChain() *store.MembershipChain {
	indirectAccountRights := &ttnpb.Rights{Rights: m.IndirectAccountRights}
	directAccountRights := &ttnpb.Rights{Rights: m.DirectAccountRights}
	c := &store.MembershipChain{
		RightsOnEntity:    directAccountRights,
		EntityIdentifiers: buildIdentifiers(m.EntityType, m.EntityFriendlyID),
	}
	switch m.DirectAccountType {
	case user:
		c.UserIdentifiers = &ttnpb.UserIdentifiers{
			UserId: m.DirectAccountFriendlyID,
		}
	case organization:
		if m.IndirectAccountType == user {
			c.UserIdentifiers = &ttnpb.UserIdentifiers{
				UserId: m.IndirectAccountFriendlyID,
			}
			c.RightsOnOrganization = indirectAccountRights
		}
		c.OrganizationIdentifiers = &ttnpb.OrganizationIdentifiers{
			OrganizationId: m.DirectAccountFriendlyID,
		}
	}
	return c
}
```

#### Checklist

- [x] Scope: The referenced issue is addressed, there are no unrelated changes.
- [x] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [x] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

**Comment by nicholaspcr** [comment]
@johanstokking since Adrian is off, I would like to ask for your review on this.
Thanks in advance

**Review by johanstokking — APPROVED** [review]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ For details about compatibility between different releases, see the **Commitment
 ### Fixed
 
 - The CLI now continues deleting devices when unclaiming from the Join Server fails. This resembles the behavior in the Console. This no longer stops devices from being deleted if the Join Server is unavailable or the claim is not held.
+- Organization API Keys' rights no longer are considered invalid during fetch operations. If the proper right is attached to said API key it is possible to fetch all fields of an entity, previous to this fix only public safe fields were fetchable.
+  - This is the default behavior but a bug that altered it was introduced on `v3.22.0`.
```
Comment (adriansmares) [inline CHANGELOG.md]: No need for the second comment.

**Review by adriansmares — APPROVED** [review]
LGTM 👍

**Pull request merged**